# Integer Division Operator

In addition to the normal divison (`/`, called the true division operator), python has other operator called the integer divison operator (`//`). 

Recall that the true interger divison operator returns the result of normal division, which could be a faction (floating point):

In [1]:
7/4

1.75

The interger division operator, unlike the true division operator, only return the integer part of from the result of true division operator:

In [2]:
7 // 4

1

Note that the result is rounds toward minus infinity or so called `flooring`: 

In [3]:
- 7 // 4

-2

In the above, the true divison of `-7` by `4` is `-1.75`. The two nearest integer of `-1.75` is `-1` and `-2`. So the result of flooring will be `-2` (because it is closer to the minus infinity).

Becareful, the truncation of interger division is not the same as rounding. For example `7//4` results in `1`; but the result of `round(7/4)` is `2`:

In [4]:
round(7/4)


2

Also, the truncation of interger division is not the same as the result of the build-in `int()` function: `-7//4` returns `2`; but `int(-7/4)` returns `-1`:

In [6]:
int(-7/4)

-1